salt/modules/aptpkg.py

```diff
@@ -652,6 +654,7 @@ def install(name=None,
         cmds.append(cmd)
 
     if to_reinstall:
+        all_pkgs.extend([x for x in six.itervalues(to_reinstall)])
```

Comment:
I don't think this is going to work. Did you test this with a specified version, or with binary packages (i.e. the ``sources`` param)? When a version is specified, The values in ``to_reinstall`` dict will be in the format ``pkgname=version``, or the path to a cached copy of the .deb package if ``sources`` was used. So, the membership check in line 677 below won't find a match in ``all_pkgs``.

This line should probably be changed to:

```python
    all_pkgs.extend(to_reinstall)
```

There should be no need for a list comprehension, as `list.extend(foo)` will extend the list by iterating over ``foo``, and iterating over a dict yields its keys, which in this case would be package names, which is what we want here.